lib/matplotlib/axes/_axes.py

```diff
@@ -3914,6 +3914,7 @@ def dopatch(xs, ys, **kwargs):
             positions = list(range(1, N + 1))
         elif len(positions) != N:
             raise ValueError(datashape_message.format("positions"))
+        positions = np.array(positions)
```

Comment:
This is a good fix for the case described in the ticket, but I think the code will continue raising obscure error when positions are not numbers. Can you add a check after this line about the dtype of the array being a number? Maybe something like:

```python
if not isinstance(positions.dtype, numbers.Number):
     raise TypeError("positions should be a iterable of dtype number, but found dtype %s" % blah blah blah.)
```

When we end up supporting something else than numbers, this can be updated.

Reply:
I couldn't quite get that check to work; I had to check the first element.  But if there is a better way, let me know.